### PR TITLE
Reduce Jekyll::Renderer instances during a build

### PR DESCRIPTION
**Pull request by ashmaroli**
- This is an 🙋 enhancement.

## Summary

This change reduces the number of `Jekyll::Renderer` instances by half

`Jekyll::Renderer` is initialized at following places in Jekyll Core:
https://github.com/jekyll/jekyll/blob/bdbf35136ebd7b9f193568bd2728721ef5d020d2/lib/jekyll/convertible.rb#L245-L247
https://github.com/jekyll/jekyll/blob/bdbf35136ebd7b9f193568bd2728721ef5d020d2/lib/jekyll/document.rb#L99-L101
https://github.com/jekyll/jekyll/blob/bdbf35136ebd7b9f193568bd2728721ef5d020d2/lib/jekyll/site.rb#L516-L521

In other words, a Renderer instance is created twice for every `Page` or `Document` instance &mdash; Once instance to get the renderer's attributes and the other instance to render the document.

### CODE REVIEW

**Comment by mattr-** [comment]
@jekyllbot merge +minor

**Comment by mattr-** [comment]
@jekyllbot: merge +minor
